hermes-management/src/main/java/pl/allegro/tech/hermes/management/domain/dc/ExceptionWrapper.java

```diff
@@ -0,0 +1,16 @@
+package pl.allegro.tech.hermes.management.domain.dc;
+
+import pl.allegro.tech.hermes.common.exception.HermesException;
+import pl.allegro.tech.hermes.common.exception.InternalProcessingException;
+
+class ExceptionWrapper {
+     static RuntimeException wrapInInternalProcessingExceptionIfNeeded(Exception toWrap,
+                                                                       String command,
+                                                                       String dcName) {
+        if (toWrap instanceof HermesException) {
+            return (HermesException) toWrap;
```

Comment:
Czy to rzutowanie jest potrzebne? Typ zwracany to ostatecznie `RuntimeException`, z którego `HermesException` dziedziczy.

Reply:
Yup, if you don't do the casting the compiler starts to complain. `toWrap` exception is passed as `Exception` argument and the type returned from the function is `RuntimeException` so they are not compatible with each other.